api/topic.go

```diff
@@ -65,6 +62,106 @@ func (api *API) getTopicPrivateHandler(w http.ResponseWriter, req *http.Request)
 	log.Event(ctx, "request successful", log.INFO, logdata) // NOTE: name of function is in logdata
 }
 
+// getSubtopicsPublicHandler is a handler that gets a topic by its id from MongoDB
+func (api *API) getSubtopicsPublicHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	id := vars["id"]
+	logdata := log.Data{
+		"request_id": ctx.Value(dprequest.RequestIdKey),
+		"topic_id":   id,
+		"function":   "getSubtopicsPublicHandler",
+	}
+
+	// get topic from mongoDB by id
+	topic, err := api.dataStore.Backend.GetTopic(id)
+	if err != nil {
+		// no topic found to retrieve the subtopics from
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	var result models.PublicSubtopics
+
+	if len(topic.Current.SubtopicIds) > 0 {
+		for _, item := range topic.Current.SubtopicIds {
+			// get sub topic from mongoDB by item
+			topic, err := api.dataStore.Backend.GetTopic(item)
```

Comment:
Instead of having to query the DB for each individual sub topic, can we make a single query that has a list of sub topic ID's?

Reply:
That may end up with more complicated code to check which ID's failed to be found in the the returned list for the list given.
It would also need a different function to do the search and require more test code.
This works, so leave as is for now.